src/snap/grpc_export.cc

```diff
@@ -48,81 +48,146 @@ using json = nlohmann::json;
+        response = http_server::response::stock_reply(
+            http_server::response::ok, this->preamble);
+    }
+
+    void log(http_server::string_type const &info) {
+        std::cerr << "Error: " << info << std::endl;
+    }
+};
+
 Plugin::GRPCExporter::GRPCExporter() : impl(std::shared_ptr<GRPCExportImpl>(implement())) {}
 
 future<void> Plugin::GRPCExporter::ExportPlugin(shared_ptr<PluginInterface> plugin, const Meta* meta) {
-  return this->impl->DoExport(plugin, meta);
+    return this->impl->DoExport(plugin, meta);
 }
 
 Plugin::GRPCExportImpl* Plugin::GRPCExporter::implement() {
-  return new GRPCExportImpl();
+    return new GRPCExportImpl();
 }
 
 future<void> Plugin::GRPCExportImpl::DoExport(shared_ptr<PluginInterface> plugin, const Meta *meta) {
-  this->plugin = std::move(plugin);
-  this->meta = meta;
-  doConfigure();
-  doRegister();
-  doAdvertise();
-  auto self = this->shared_from_this();
-  return std::async(std::launch::deferred, [=](){ self->doJoin(); });
+    this->plugin = std::move(plugin);
+    this->meta = meta;
+    doConfigure();
+    doRegister();
+    //_preamble = printPreamble();
+
+    if (this->meta->stand_alone) {
+        auto start_sa = std::async(std::launch::deferred, &Plugin::GRPCExportImpl::start_stand_alone, this,
+                                    this->meta->stand_alone_port);
+        start_sa.get();
+    }
+    else cout << printPreamble() << endl;
+    
+    auto self = this->shared_from_this();
+    return std::async(std::launch::deferred, [=](){ self->doJoin(); });
 }
 
 void Plugin::GRPCExportImpl::doConfigure() {
-  string server_address = "127.0.0.1:0";
-
-  switch (plugin->GetType()) {
-    case Plugin::Collector:
-      this->service.reset(new Proxy::CollectorImpl(plugin->IsCollector()));
-      break;
-    case Plugin::Processor:
-      this->service.reset(new Proxy::ProcessorImpl(plugin->IsProcessor()));
-      break;
-    case Plugin::Publisher:
-      this->service.reset(new Proxy::PublisherImpl(plugin->IsPublisher()));
-      break;
-  }
-  builder.reset(new grpc::ServerBuilder());
-  builder->AddListeningPort(server_address, grpc::InsecureServerCredentials(),
-                           &this->port);
-
+    std::stringstream ss;
+    ss << this->meta->listen_addr << ":";
+    this->meta->listen_port == "" ? ss << "0" : ss << this->meta->listen_port;
+
+    //string server_address = "127.0.0.1:0";
+
+    switch (plugin->GetType()) {
+        case Plugin::Collector:
+            this->service.reset(new Proxy::CollectorImpl(plugin->IsCollector()));
+            break;
+        case Plugin::Processor:
+            this->service.reset(new Proxy::ProcessorImpl(plugin->IsProcessor()));
+            break;
+        case Plugin::Publisher:
+            this->service.reset(new Proxy::PublisherImpl(plugin->IsPublisher()));
+            break;
+        //TODO: aarontay
+        //case Plugin::StreamCollector
+        //
+        // break;
+        //default:
```

Comment:
default case could be useful, please consider to uncomment this part or remove it